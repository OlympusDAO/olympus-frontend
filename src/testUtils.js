import React from "react";
import { render } from "@testing-library/react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";
import { ThemeProvider } from "@material-ui/core/styles";
import CssBaseline from "@material-ui/core/CssBaseline";
import { light as lightTheme } from "./themes/light.js";
import { Web3ContextProvider } from "./hooks/web3Context";
import store from "./store.ts";
import { createMemoryHistory } from "history";
import App from "./App";
import { I18nProvider } from "@lingui/react";
import { i18n } from "@lingui/core";

const AllTheProviders = ({ children }) => {
  return (
    <Web3ContextProvider>
      <Provider store={store}>
        <BrowserRouter basename={"/"}>
          <ThemeProvider theme={lightTheme}>
            <CssBaseline />
            {children}
          </ThemeProvider>
        </BrowserRouter>
      </Provider>
    </Web3ContextProvider>
  );
};

const customRender = (ui, options) => render(ui, { wrapper: AllTheProviders, ...options });

const renderRoute = function (route) {
  const history = createMemoryHistory();
  history.push(route);
  render(
    <Web3ContextProvider>
      <Provider store={store}>
        <I18nProvider i18n={i18n}>
          <BrowserRouter basename={"/#"}>
            <App />
          </BrowserRouter>
        </I18nProvider>
      </Provider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
