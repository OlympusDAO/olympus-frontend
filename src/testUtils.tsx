import { i18n } from "@lingui/core";
import { I18nProvider } from "@lingui/react";
import CssBaseline from "@material-ui/core/CssBaseline";
import { ThemeProvider } from "@material-ui/core/styles";
import { render, RenderOptions, RenderResult } from "@testing-library/react";
import { createMemoryHistory } from "history";
import React, { ReactElement, ReactNode } from "react";
import { Provider } from "react-redux";
import { BrowserRouter } from "react-router-dom";

import App from "./App";
import { Web3ContextProvider } from "./hooks/web3Context";
import { ReactQueryProvider } from "./lib/react-query";
import defaultStore from "./store";
import { light as lightTheme } from "./themes/light.js";

const customRender = (ui: ReactElement, store = defaultStore, options?: RenderOptions): RenderResult => {
  const ProviderWrapper = ({ children }: { children?: ReactNode }) => (
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <ThemeProvider theme={lightTheme}>
                <CssBaseline />
                {children}
              </ThemeProvider>
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>
  );
  return render(ui, { wrapper: ProviderWrapper, ...options });
};

const renderRoute = function (route: string, store = defaultStore) {
  const history = createMemoryHistory();
  history.push(route);
  return render(
    <Web3ContextProvider>
      <ReactQueryProvider>
        <Provider store={store}>
          <I18nProvider i18n={i18n}>
            <BrowserRouter basename={"/"}>
              <App />
            </BrowserRouter>
          </I18nProvider>
        </Provider>
      </ReactQueryProvider>
    </Web3ContextProvider>,
  );
};

// re-export everything
export * from "@testing-library/react";

// override render method
export { customRender as render, renderRoute };
