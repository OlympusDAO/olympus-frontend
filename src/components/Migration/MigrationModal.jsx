import {
  Backdrop,
  Box,
  Button,
  ButtonBase,
  Fade,
  Modal,
  SvgIcon,
  Table,
  TableCell,
  TableHead,
  TableRow,
  TableBody,
  Typography,
  Paper,
} from "@material-ui/core";
import { NetworkID } from "../../lib/Bond";
console.log(NetworkID);
import { addresses } from "../../constants";

// import ButtonUnstyled from "@mui/core/ButtonUnstyled";
import { ReactComponent as XIcon } from "../../assets/icons/x.svg";
import { makeStyles } from "@material-ui/core/styles";
import { useDispatch, useSelector } from "react-redux";
import { BigNumber } from "ethers";
import { changeMigrationApproval, migrateAll } from "src/slices/MigrateThunk";
import { useWeb3Context } from "src/hooks";
import { useEffect, useMemo } from "react";
import { isPendingTxn, txnButtonText } from "src/slices/PendingTxnsSlice";
import { info } from "src/slices/MessagesSlice";
import InfoTooltip from "../InfoTooltip/InfoTooltip";
import "./migration-modal.scss";
import { useAppSelector } from "src/hooks";
import { trim } from "src/helpers";
const formatCurrency = c => {
  return new Intl.NumberFormat("en-US", {
    style: "currency",
    currency: "USD",
    maximumFractionDigits: 0,
    minimumFractionDigits: 0,
  }).format(c);
};
const style = {
  position: "absolute",
  top: "50%",
  left: "50%",
  transform: "translate(-50%, -50%)",
  bgcolor: "background.paper",
  boxShadow: 24,
  p: 2,
  zIndex: 3,
  maxWidth: 600,
  minWidth: 300,
  borderRadius: 10,
};

const useStyles = makeStyles({
  custom: {
    color: "#00EE00",
  },
});

function MigrationModal({ open, handleOpen, handleClose }) {
  const dispatch = useDispatch();
  const classes = useStyles();
  const { provider, address, connect } = useWeb3Context();

  const pendingTransactions = useSelector(state => {
    return state.pendingTransactions;
  });

  let rows = [];
  let isMigrationComplete = useSelector(state => state.account.isMigrationComplete);
  const networkId = NetworkID;
  // console.log(networkId);
  const onSeekApproval = token => {
    dispatch(
      changeMigrationApproval({
        address,
        networkID: networkId,
        provider,
        token: token.toLowerCase(),
        displayName: token,
        insertName: true,
      }),
    );
  };

  const onMigrate = () => dispatch(migrateAll({ provider, address, networkID: networkId }));
  const currentIndex = useSelector(state => state.app.currentIndex);

  const currentOhmBalance = useSelector(state => Number(state.account.balances.ohm));
  const currentSOhmBalance = useSelector(state => Number(state.account.balances.sohm));
  const currentWSOhmBalance = useSelector(state => Number(state.account.balances.wsohm));
  const wsOhmPrice = useSelector(state => state.app.marketPrice * state.app.currentIndex);

  const marketPrice = useSelector(state => {
    return state.app.marketPrice;
  });
  const approvedOhmBalance = useSelector(state => Number(state.account.migration.ohm));
  const approvedSOhmBalance = useSelector(state => Number(state.account.migration.sohm));
  const approvedWSOhmBalance = useSelector(state => Number(state.account.migration.wsohm));
  const ohmFullApproval = approvedOhmBalance >= currentOhmBalance;
  const sOhmFullApproval = approvedSOhmBalance >= currentSOhmBalance;
  const wsOhmFullApproval = approvedWSOhmBalance >= currentWSOhmBalance;
  const isAllApproved = ohmFullApproval && sOhmFullApproval && wsOhmFullApproval;

  const ohmInUSD = formatCurrency(marketPrice * currentOhmBalance);
  const sOhmInUSD = formatCurrency(marketPrice * currentSOhmBalance);
  const wsOhmInUSD = formatCurrency(wsOhmPrice * currentWSOhmBalance);

  useEffect(() => {
    if (isAllApproved && (currentOhmBalance || currentSOhmBalance || currentWSOhmBalance)) {
      dispatch(info("All approvals complete. You may now migrate."));
    }
  }, [isAllApproved]);

  rows = [
    {
      initialAsset: "OHM",
      initialBalance: currentOhmBalance,
      targetAsset: "gOHM",
      targetBalance: currentOhmBalance / currentIndex,
      fullApproval: ohmFullApproval,
      usdBalance: ohmInUSD,
    },
    {
      initialAsset: "sOHM",
      initialBalance: currentSOhmBalance,
      targetAsset: "gOHM",
      targetBalance: currentSOhmBalance / currentIndex,
      fullApproval: sOhmFullApproval,
      usdBalance: sOhmInUSD,
    },
    {
      initialAsset: "wsOHM",
      initialBalance: currentWSOhmBalance,
      targetAsset: "gOHM",
      targetBalance: currentWSOhmBalance,
      fullApproval: wsOhmFullApproval,
      usdBalance: wsOhmInUSD,
    },
  ];

  return (
    <div>
      <Modal
        className="mig-modal-full"
        aria-labelledby="migration-modal-title"
        aria-describedby="migration-modal-description"
        open={open}
        onClose={handleClose}
        closeAfterTransition
        BackdropComponent={Backdrop}
        BackdropProps={{
          timeout: 500,
        }}
      >
        <Fade in={open}>
          <Box sx={style} className="modal-inner">
            <Box display="flex" flexDirection="row" alignItems="center" justifyContent="space-between">
              <Button onClick={handleClose}>
                <SvgIcon component={XIcon} color="primary" />
              </Button>
              <Box paddingRight={6}>
                <Typography id="migration-modal-title" variant="h6" component="h2">
                  {isMigrationComplete
                    ? "Migration complete"
                    : isAllApproved
                    ? "You are now ready to migrate"
                    : "You have assets ready to migrate to v2"}
                </Typography>
              </Box>
              <Box />
            </Box>
            {isMigrationComplete ? null : (
              <Box paddingTop={4}>
                <Typography id="migration-modal-description" variant="body1">
                  {isAllApproved
                    ? "Click on the Migrate button to complete the upgrade to v2. "
                    : "Olympus v2 introduces upgrades to on-chain governance and bonds to enhance decentralization and immutability. "}
                  <a
                    href="https://docs.olympusdao.finance/main/basics/migration"
                    target="_blank"
                    color="inherit"
                    rel="noreferrer"
                    className="docs-link"
                  >
                    <u>Learn More</u>
                  </a>
                </Typography>
              </Box>
            )}

            <Table>
              <TableHead>
                <TableRow>
                  <TableCell align="center">
                    <Typography>Asset</Typography>
                  </TableCell>
                  <TableCell align="center">
                    <Box display="inline-flex">
                      <Typography>Pre-migration</Typography>
                      <InfoTooltip
                        className="migartion-tooltip"
                        message={"This is the current balance of v1 assets in your wallet."}
                      ></InfoTooltip>
                    </Box>
                  </TableCell>
                  <TableCell align="center">
                    <Box display="inline-flex">
                      <Typography>Post-migration</Typography>
                      <InfoTooltip
                        className="migartion-tooltip"
                        message={
                          "This is the equivalent amount of gOHM you will have in your wallet once migration is complete."
                        }
                      ></InfoTooltip>
                    </Box>
                  </TableCell>

                  <TableCell align="center">
                    <Box display="inline-flex">{/* <Typography>Migration Completion Status</Typography> */}</Box>
                  </TableCell>

                  <TableCell align="left"></TableCell>
                </TableRow>
              </TableHead>
              <TableBody>
                {rows
                  .filter(asset => asset.initialBalance > 0)
                  .map(row => (
                    <TableRow key={row.initialAsset} sx={{ "&:last-child td, &:last-child th": { border: 0 } }}>
                      <TableCell component="th" scope="row">
                        <Typography>{`${row.initialAsset} -> ${row.targetAsset}`}</Typography>
                      </TableCell>
                      <TableCell align="left">
                        <Typography>
                          {trim(row.initialBalance, 4)} {row.initialAsset}
                          <Typography style={{ marginTop: "10px" }}>{`(${row.usdBalance})`}</Typography>
                        </Typography>
                      </TableCell>
                      <TableCell align="left">
                        <Typography>
                          {trim(row.targetBalance, 4)} {row.targetAsset}
                          <Typography style={{ marginTop: "10px" }}>{`(${row.usdBalance})`}</Typography>
                        </Typography>
                      </TableCell>
                      <TableCell align="left">
                        {isMigrationComplete ? (
                          <Typography align="center" className={classes.custom}>
                            Migrated
                          </Typography>
                        ) : row.fullApproval ? (
                          <Typography align="center" className={classes.custom}>
                            Approved
                          </Typography>
                        ) : (
                          <Button
                            variant="outlined"
                            onClick={() => onSeekApproval(row.initialAsset)}
                            disabled={isPendingTxn(
                              pendingTransactions,
                              `approve_migration_${row.initialAsset.toLowerCase()}`,
                            )}
                          >
                            <Typography>
                              {txnButtonText(
                                pendingTransactions,
                                `approve_migration_${row.initialAsset.toLowerCase()}`,
                                "Approve",
                              )}
                            </Typography>
                          </Button>
                        )}
                      </TableCell>
                    </TableRow>
                  ))}
              </TableBody>
            </Table>

            <Box display="flex" flexDirection="row" justifyContent="center">
              <Button
                color="primary"
                variant="contained"
                disabled={!isAllApproved || isPendingTxn(pendingTransactions, "migrate_all")}
                onClick={isMigrationComplete ? handleClose : onMigrate}
              >
                <Box marginX={4} marginY={0.5}>
                  <Typography>
                    {isMigrationComplete ? "Close" : txnButtonText(pendingTransactions, "migrate_all", "Migrate")}
                  </Typography>
                </Box>
              </Button>
            </Box>
          </Box>
        </Fade>
      </Modal>
    </div>
  );
}

export default MigrationModal;
