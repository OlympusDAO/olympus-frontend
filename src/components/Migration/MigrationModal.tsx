import "./MigrationModal.scss";

import { t, Trans } from "@lingui/macro";
import { Box, Button, Table, TableBody, TableCell, TableHead, TableRow, Typography } from "@mui/material";
import useMediaQuery from "@mui/material/useMediaQuery";
import { InfoTooltip, Modal, Tab, Tabs } from "@olympusdao/component-library";
import { useEffect } from "react";
import { useDispatch } from "react-redux";
import { NetworkId } from "src/constants";
import { trim } from "src/helpers";
import { useMigrationData } from "src/helpers/Migration";
import { useWeb3Context } from "src/hooks";
import { useAppSelector } from "src/hooks";
import { info } from "src/slices/MessagesSlice";
import { changeMigrationApproval, migrateAll } from "src/slices/MigrateThunk";
import { isPendingTxn, txnButtonText } from "src/slices/PendingTxnsSlice";
import { AppDispatch } from "src/store";

const classes = {
  custom: {
    color: "#00EE00",
  },
};

function MigrationModal({ open, handleClose }: { open: boolean; handleClose: any }) {
  const dispatch: AppDispatch = useDispatch();
  const isMobileScreen = useMediaQuery("(max-width: 513px)");
  const { provider, address, networkId } = useWeb3Context();

  const {
    view,
    changeView,
    currentOhmBalance,
    currentSOhmBalance,
    currentWSOhmBalance,
    ohmFullApproval,
    sOhmFullApproval,
    wsOhmFullApproval,
    ohmAsgOHM,
    sOHMAsgOHM,
    ohmInUSD,
    sOhmInUSD,
    wsOhmInUSD,
    isGOHM,
    targetAsset,
    targetMultiplier,
    oldAssetsDetected,
    pendingTransactions,
    isAllApproved,
  } = useMigrationData();

  let rows: any[] = [];
  const isMigrationComplete = useAppSelector(state => state.account.isMigrationComplete);

  const onSeekApproval = (token: string) => {
    dispatch(
      changeMigrationApproval({
        address,
        networkID: networkId,
        provider,
        token: token.toLowerCase(),
        displayName: token,
        insertName: true,
      }),
    );
  };

  useEffect(() => {
    if (
      networkId &&
      (networkId === NetworkId.MAINNET || networkId === NetworkId.TESTNET_RINKEBY) &&
      isAllApproved &&
      (currentOhmBalance || currentSOhmBalance || currentWSOhmBalance)
    ) {
      dispatch(info("All approvals complete. You may now migrate."));
    }
  }, [isAllApproved]);

  const onMigrate = () => dispatch(migrateAll({ provider, address, networkID: networkId, gOHM: isGOHM }));

  rows = [
    {
      initialAsset: "OHM",
      initialBalance: currentOhmBalance,
      targetAsset: targetAsset,
      targetBalance: ohmAsgOHM * targetMultiplier,
      fullApproval: ohmFullApproval,
      usdBalance: ohmInUSD,
    },
    {
      initialAsset: "sOHM",
      initialBalance: currentSOhmBalance,
      targetAsset: targetAsset,
      targetBalance: sOHMAsgOHM * targetMultiplier,
      fullApproval: sOhmFullApproval,
      usdBalance: sOhmInUSD,
    },
    {
      initialAsset: "wsOHM",
      initialBalance: currentWSOhmBalance,
      targetAsset: targetAsset,
      targetBalance: +currentWSOhmBalance * targetMultiplier,
      fullApproval: wsOhmFullApproval,
      usdBalance: wsOhmInUSD,
    },
  ];

  // console.info(`MigrationModal rows after: [${rows}]`);

  return (
    <div data-testid="migration-modal">
      <Modal
        aria-labelledby="migration-modal-title"
        aria-describedby="migration-modal-description"
        open={open}
        onClose={handleClose}
        closeAfterTransition
        minHeight={"200px"}
        closePosition={"left"}
        headerText={
          isMigrationComplete || !oldAssetsDetected
            ? t`Migration complete`
            : isAllApproved
            ? t`You are now ready to migrate`
            : t`You have assets ready to migrate to v2`
        }
      >
        <>
          {isMigrationComplete || !oldAssetsDetected ? null : (
            <Box paddingTop={isMobileScreen ? 2 : 4} paddingBottom={isMobileScreen ? 2 : 0}>
              <Typography id="migration-modal-description" variant="body2" className={isMobileScreen ? `mobile` : ``}>
                {isAllApproved
                  ? t`Click on the Migrate button to complete the upgrade to v2.`
                  : t`Olympus v2 introduces upgrades to on-chain governance and bonds to enhance decentralization and immutability.`}{" "}
                <a
                  href="https://docs.olympusdao.finance/main/basics/migration"
                  target="_blank"
                  color="inherit"
                  rel="noreferrer"
                  className="docs-link"
                >
                  <u>
                    <Trans>Learn More</Trans>
                  </u>
                </a>
              </Typography>
            </Box>
          )}
          <Box display="flex" justifyContent="center" marginTop={1}>
            <Typography variant="h5" color="textSecondary">
              <Trans>Migration Output</Trans>
            </Typography>
          </Box>
          <Tabs
            centered
            value={view}
            textColor="primary"
            indicatorColor="primary"
            onChange={changeView}
            aria-label="payout token tabs"
            className="payout-token-tabs"
          >
            <Tab aria-label="payout-sohm-button" label="sOHM" className="payout-token-tab" />
            <Tab aria-label="payout-gohm-button" label="gOHM" className="payout-token-tab" />
          </Tabs>
          {isMobileScreen ? (
            <Box id="mobile-container-migration">
              {rows
                .filter(asset => asset.initialBalance > 0)
                .map(row => (
                  <Box style={{ margin: "20px 0px 20px 0px" }}>
                    <Typography
                      id="m-asset-row"
                      style={{ margin: "10px 0px 10px 0px", fontWeight: 700 }}
                    >{`${row.initialAsset} -> ${row.targetAsset}`}</Typography>
                    <Box display="flex" flexDirection="row" justifyContent="space-between">
                      <Typography>
                        {trim(row.initialBalance, 4)} {row.initialAsset}
                      </Typography>
                      <Typography>{`(${row.usdBalance})`}</Typography>
                    </Box>
                    <Box display="flex" justifyContent="center" style={{ margin: "10px 0px 10px 0px" }}>
                      {isMigrationComplete || !oldAssetsDetected ? (
                        <Typography align="center" sx={classes.custom}>
                          <Trans>Migrated</Trans>
                        </Typography>
                      ) : row.fullApproval ? (
                        <Typography align="center" sx={classes.custom}>
                          <Trans>Approved</Trans>
                        </Typography>
                      ) : (
                        <Button
                          variant="outlined"
                          onClick={() => onSeekApproval(row.initialAsset)}
                          aria-label="approve-migration"
                          disabled={isPendingTxn(
                            pendingTransactions,
                            `approve_migration_${row.initialAsset.toLowerCase()}`,
                          )}
                        >
                          <Typography>
                            {txnButtonText(
                              pendingTransactions,
                              `approve_migration_${row.initialAsset.toLowerCase()}`,
                              t`Approve`,
                            )}
                          </Typography>
                        </Button>
                      )}
                    </Box>
                  </Box>
                ))}
            </Box>
          ) : (
            <Table>
              <TableHead>
                <TableRow style={{ verticalAlign: "top" }}>
                  <TableCell align="center">
                    <Typography>Asset</Typography>
                  </TableCell>
                  <TableCell align="left">
                    <Box display="flex">
                      <Box display="inline-flex">
                        <Typography>
                          <Trans>Pre-migration</Trans>
                        </Typography>
                        <InfoTooltip
                          message={t`This is the current balance of v1 assets in your wallet.`}
                          children={undefined}
                        />
                      </Box>
                    </Box>
                  </TableCell>
                  <TableCell align="left">
                    <Box display="flex" flexDirection="column">
                      <Box display="inline-flex">
                        <Typography>
                          <Trans>Post-migration</Trans>
                        </Typography>
                        <InfoTooltip
                          message={t`This is the equivalent amount of gOHM you will have in your wallet once migration is complete.`}
                          children={undefined}
                        />
                      </Box>
                    </Box>
                    <Box display="inline-flex">
                      <Typography variant="body2">
                        <Trans>(includes rebase rewards)</Trans>
                      </Typography>
                    </Box>
                  </TableCell>

                  <TableCell align="center">
                    <Box display="inline-flex">{/* <Typography>Migration Completion Status</Typography> */}</Box>
                  </TableCell>

                  <TableCell align="left" />
                </TableRow>
              </TableHead>
              <TableBody>
                {rows
                  .filter(asset => asset.initialBalance > 0)
                  .map(row => (
                    <TableRow key={row.initialAsset}>
                      <TableCell component="th" scope="row">
                        <Typography>{`${row.initialAsset} -> ${row.targetAsset}`}</Typography>
                      </TableCell>
                      <TableCell align="left">
                        <Typography>
                          {trim(row.initialBalance, 4)} {row.initialAsset}
                          <Typography style={{ marginTop: "10px" }}>{`(${row.usdBalance})`}</Typography>
                        </Typography>
                      </TableCell>
                      <TableCell align="left">
                        <Typography>
                          {trim(row.targetBalance, 4)} {row.targetAsset}
                          <Typography style={{ marginTop: "10px" }}>{`(${row.usdBalance})`}</Typography>
                        </Typography>
                      </TableCell>
                      <TableCell align="left">
                        {isMigrationComplete || !oldAssetsDetected ? (
                          <Typography align="center" sx={classes.custom}>
                            <Trans>Migrated</Trans>
                          </Typography>
                        ) : row.fullApproval ? (
                          <Typography align="center" sx={classes.custom}>
                            <Trans>Approved</Trans>
                          </Typography>
                        ) : (
                          <Button
                            variant="outlined"
                            onClick={() => onSeekApproval(row.initialAsset)}
                            disabled={isPendingTxn(
                              pendingTransactions,
                              `approve_migration_${row.initialAsset.toLowerCase()}`,
                            )}
                          >
                            <Typography>
                              {txnButtonText(
                                pendingTransactions,
                                `approve_migration_${row.initialAsset.toLowerCase()}`,
                                t`Approve`,
                              )}
                            </Typography>
                          </Button>
                        )}
                      </TableCell>
                    </TableRow>
                  ))}
              </TableBody>
            </Table>
          )}

          <Box display="flex" flexDirection="row" justifyContent="center">
            <Button
              color="primary"
              variant="contained"
              disabled={!isAllApproved || isPendingTxn(pendingTransactions, "migrate_all")}
              onClick={isMigrationComplete || !oldAssetsDetected ? handleClose : onMigrate}
              fullWidth={isMobileScreen}
            >
              <Box marginX={4} marginY={0.5}>
                <Typography>
                  {isMigrationComplete || !oldAssetsDetected
                    ? "Close"
                    : txnButtonText(pendingTransactions, "migrate_all", t`Migrate all to ${isGOHM ? "gOHM" : "sOHM"}`)}
                </Typography>
              </Box>
            </Button>
          </Box>
          <div className="help-text">
            <em>
              <Typography variant="body2" style={isMobileScreen ? { lineHeight: "1em" } : {}}>
                <Trans>
                  Save on gas fees by migrating all your assets to the new gOHM or sOHM in one transaction. Each asset
                  asset above must be approved before all can be migrated.
                </Trans>
              </Typography>
            </em>
          </div>
        </>
      </Modal>
    </div>
  );
}

export default MigrationModal;
