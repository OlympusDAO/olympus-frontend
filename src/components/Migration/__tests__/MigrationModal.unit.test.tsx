import { render } from "../../../testUtils";
import MigrationModal from "../MigrationModal";

describe("<MigrationModal/>", () => {
  it("should render component", () => {
    const { container } = render(<MigrationModal open={false} handleClose={() => console.log("handleClose")} />);
    expect(container).toMatchSnapshot();
  });
});
