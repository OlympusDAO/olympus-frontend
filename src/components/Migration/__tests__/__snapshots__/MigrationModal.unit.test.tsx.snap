// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<MigrationModal/> should render component 1`] = `
<div>
  <div />
</div>
`;
